packages/flutter/lib/src/widgets/_web_image_web.dart

```diff
@@ -0,0 +1,241 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:js_interop';
+import 'dart:ui_web' as ui_web;
+
+import '../../rendering.dart';
+import '../../widgets.dart';
+import '../painting/_network_image_web.dart';
+import '../web.dart' as web;
+
+/// Returns [true] if the bytes at [url] can be fetched.
+///
+/// The bytes may be unable to be fetched if they aren't from the same origin
+/// and the server hosting them does not allow cross-origin requests.
+Future<bool> checkIfImageBytesCanBeFetched(String url) async {
+  // First check if url is same origin.
+  final Uri uri = Uri.parse(url);
+  if (uri.origin == web.window.origin) {
+    return true;
+  }
+
+  final Completer<web.XMLHttpRequest> completer =
+      Completer<web.XMLHttpRequest>();
+  final web.XMLHttpRequest request = httpRequestFactory();
+
+  request.open('HEAD', url, true);
+
+  request.addEventListener(
+      'load',
+      (web.Event e) {
+        final int status = request.status;
+        final bool accepted = status >= 200 && status < 300;
+        final bool fileUri = status == 0; // file:// URIs have status of 0.
+        final bool notModified = status == 304;
+        final bool unknownRedirect = status > 307 && status < 400;
+        final bool success =
+            accepted || fileUri || notModified || unknownRedirect;
+
+        if (success) {
+          completer.complete(request);
+        } else {
+          completer.completeError(e);
+        }
+      }.toJS);
+
+  request.addEventListener(
+      'error', ((JSObject e) => completer.completeError(e)).toJS);
+
+  request.send();
+
+  return completer.future.then((_) {
+    return true;
+  }, onError: (_) {
+    return false;
+  });
+}
+
+/// Returns a widget which displays the [src] in an <img> tag.
+Widget createImgElementWidget(String src,
+        {Key? key,
```

Comment:
The formatting of the parameter list is a bit off, due to a missing trailing comma. You might also want to use a block body?

Reply:
Done.